src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period " + dateFrom + " - " + dateTo
+                + System.lineSeparator());
+        LocalDate dateFromFormatted = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToFormatted = LocalDate.parse(dateTo, formatter);
+        for (String name : names) {
+            result.append(formatData(name, calculatePersonsIncome(name, data,
+                    dateFromFormatted, dateToFormatted)));
```

Comment:
Has bad readable method inside methods parameters, write it more simple using more variables